src/cpu.jakt

```diff
@@ -349,216 +349,201 @@ class CPU {
         .test_nz_flags(value: .a)
     }
 
-    function adc(mut this, opcode: u8) {
-        let carry_value = match .carry {
-            true => 1i16
-            else => 0i16
+    function adder(mut this, operand: u8, carry: bool) -> u8 {
+        // TODO: decimal mode
+        let a = .a as! u16;
+        let b = operand as! u16;
+        let c = match carry { 
+            true => 1u16
+            else => 0u16
         }
+        let intermediate = a + b + c
+        .carry = intermediate > 0xFF
+        let result = intermediate as! u8
```

Comment:
This line crashes with `Assertion failed: (false && "Jakt::is_within_range<OutputType>(input)"), function operator()` intermediate is greater than 0xff. Changing to `let result = (intermediate & 0xff) as! u8` fixes this issue.

But using .adder() for CMP is an issue, as it will update the overflow flag, which CMP should not do.

Reply:
Thanks. I overlooked the overflow flag + CMP detail.

Reply:
I'm also learning `jakt`, so I used `as_truncated` to explicitly cast from `u16` to `u8` instead of masking + `as!`.